doc/specs/#182 - Support for installation of portable standalone apps.md

```diff
@@ -0,0 +1,213 @@
+The Product Code will be populated with the "PackageIdentifier" if it is not specified in the “AppsAndFeaturesEntry”.
+
+No shortcuts or icons will be created for the package.
+
+>Note: Some portable programs may require dependencies. Dependencies are not covered in this specification and are not yet supported as of the writing of this specification.
+
+#### Scope Behavior
+
+By default, portable apps will be installed with the "User" scope unless specified otherwise by the user through the "--scope" argument or in their settings. This also means that the "scope" field in the manifest will be ignored when installing portable apps. Since portable apps are unique in that they are simply standalone executables that are copied to a specific install location based on scope, it does not seem reasonable to have the manifest enforce an install behavior that is not required. Manifest validation would also need to be updated to show an error to the user that the "scope" field does not apply if a portable installer type is specified.
+
+#### Installation from multiple sources:
+
+If the user chooses to install the same package but from a secondary source, the Windows Package Manager will create an additional subdirectory to indicate the source that the package came from. Since entries in the registry must be unique, if the entry in the "App Paths" registry conflicts with an existing entry, then the user will be notified of the conflict and the subkey value will be updated to reflect the change in executable path. In other words, the latest installed package will take precedence. This will require utilizing the installed packages data store to determine whether a package is being installed from two different sources.
+
+
+### Upgrade
+
+The package is upgraded in the same path as the installed version. The first step the Windows Package Manager will perform is to download the executable to a temporary location, and attempt to copy the exe to the specified install location. If an exe with the same name already exists, the Windows Package Manager will attempt to overwrite the file. If that process fails because the file is currently in use, the user will be informed the package is running so they can shut it down. Optionally, the user may specify "--force" to forcefully shut the application down for upgrade. Once the exe has successfully been copied to the specified install location, the "App Paths" registry and the entry in "Apps & Features" will be updated accordingly.
+
+If the "UninstallPrevious" field is specified in the manifest, then the Windows Package Manager will perform an uninstall of the previous version of the package prior to installing the newer version. 
+
+A new Windows Apps & Features entry is created to correctly report the upgraded version for future potential upgrades.
```

Comment:
```suggestion
A new Windows Apps & Features entry is created to correctly report the upgraded version for future potential upgrades.
```
Is this a new entry, or is the existing entry from the existing package updated?

Reply:
If we uninstall, then the entry is removed. If we can simply upgrade, the existing entry should get updated.